### PR TITLE
ENH/WIP: use scipy for psds + spectral plots.

### LINKED ISSUE #1743: ENH/FIX: use scipy / avoid creating figure for computing PSDs
mpl.psd does not play nicely with memory / forking / joblib. Use matplotlib.mlab.csd directly to avoid opening figures or use scipy.signal.welch instead.

example:

```Python

def _pwelch(epoch, nperseg, noverlap, nfft, fs):
    return [welch(channel, nperseg=nperseg, noverlap=noverlap,
                  nfft=nfft, fs=fs)[1]
            for channel in epoch]


def compute_psds(epochs, fmin=0, fmax=np.inf,
                 segment_size=128, n_overlap=100, n_fft=4096, n_jobs=1,
                 keep_mean=False, verbose=None):
    X = epochs.get_data()
    if keep_mean is not True:
        X = X.copy()
        X -= np.nanmean(X, axis=-1)[..., np.newaxis]

    sfreq = epochs.info['sfreq']
    parallel, my_pwelch, _ = parallel_func(
        _pwelch, n_jobs=n_jobs, verbose=verbose)

    psds = np.empty(epochs.get_data().shape[:-1] + (n_fft // 2 + 1,))
    freqs = np.arange(psds.shape[-1]) * (sfreq / n_fft)
    for i_epoch, fepoch in enumerate(parallel(
        my_pwelch(epoch, nperseg=segment_size, noverlap=n_overlap, nfft=n_fft,
                  fs=epochs.info['sfreq']) for epoch in X)):
        for i_channel, fchannel in enumerate(fepoch):
            psds[i_epoch, i_channel, :] = fchannel

    freq_mask = np.where(np.logical_and((freqs > fmin), (freqs < fmax)))[0]
    freqs = freqs[freq_mask]
    psds = psds[..., freq_mask]
    return psds, freqs
```

### PR DESCRIPTION
**Pull request by dengemann**
closes #1743 and #1856 .

### CODE REVIEW

**Comment by dengemann** [comment]
Example of updated + normalized spectral plot:

screenshot 2015-03-31 03 33 27

**Comment by agramfort** [comment]
@dengemann are you happy here?

**Comment by dengemann** [comment]
@agramfort no I'm not. There are some bugs with picking for combined MEG/EEG datasets and a few more things to tweek. I need another iteration at least.

**Comment by agramfort** [comment]
ok let me know when you had time to do this.

**Comment by dengemann** [comment]
> ok let me know when you had time to do this.

Next time I'm bored :)

**Comment by dengemann** [comment]
In fact I want to merge this one soon, so don't you worry.

**Comment by dengemann** [comment]
@yousrabk @Eric89GXL @agramfort I think I'm happy here. Please feel free to review.

File: mne/viz/topomap.py [hunk 1]
```diff
@@ -1067,3 +1068,234 @@ def plot_evoked_topomap(evoked, times=None, ch_type='mag', layout=None,
...
+    Returns
+    -------
+    fig : instance of matplotlib figure
+        Figure distributing one image per channel across sensor topography.
+    """
+
+    info = epochs.info
+    if ch_type is None:
+        ch_type = 'mag' if 'meg' in epochs else 'eeg'
+    if layout is None and ch_type is not 'eeg':
```
Comment (Eric89GXL) [inline mne/viz/topomap.py]: use `!=` instead of `is not` for str comp

File: mne/viz/topomap.py [hunk 2]
```diff
@@ -1067,3 +1068,234 @@ def plot_evoked_topomap(evoked, times=None, ch_type='mag', layout=None,
...
+        cbar.set_ticks((vmin, vmax))
+        if unit is not None:
+            cbar.ax.set_title(unit, fontsize=8)
+        cbar.ax.tick_params(labelsize=8)
+
+
+def plot_epochs_psd_topomap(epochs, bands=None, vmin=None, vmax=None,
+                            proj=False, n_fft=256, ch_type=None,
+                            n_overlap=0, layout=None,
+                            cmap='RdBu_r', agg_fun=np.mean, dB=False, n_jobs=1,
```
Comment (Eric89GXL) [inline mne/viz/topomap.py]: shouldn't this `agg_fun` also be `None` by default like the other? in any case, it should be consistent

Comment (dengemann) [inline mne/viz/topomap.py]: yes!

**Comment by Eric89GXL** [comment]
I looked through your new comments individually and other than my minor comments, LGTM +1 for merge if the CIs are happy

**Comment by dengemann** [comment]
Updated what's new. Ready to merge!

**Comment by agramfort** [comment]
looks like you need to rebase

**Comment by agramfort** [comment]
+1 for merge.

fix the merge conflict and merge.

thanks @dengemann !

**Comment by dengemann** [comment]
> +1 for merge.

Btw. I don't combine grads here (which seems to be correct when considering power).

**Comment by agramfort** [comment]
How does it work then as grads pairs share locations?

**Comment by dengemann** [comment]
> How does it work then as grads pairs share locations?
see the plots above? 202 grads

**Comment by agramfort** [comment]
travis is not happy.

to be honest I don't understand how this can fly with non combined grads. 

does it produce nice results on relevant data with TF content?

**Comment by dengemann** [comment]
See latest commits alex, merged the grads

**Comment by dengemann** [comment]
@yousrabk @Eric89GXL thanks for assisting me with this, you both helped greatly improve my initial solution. Nice MNE teamwork!

**Pull request merged**